### PR TITLE
Show categories and tags on each post

### PR DESCRIPTION
**Pull request by hszhakka**
This PR plans to implement a category archive page and a tag archive page.

### CODE REVIEW

File: _includes/post-categories.html [hunk 1]
```diff
@@ -0,0 +1,10 @@
+<span class="post-categories">{%- if page.categories.size == 1 -%}Category{%- else -%}Categories{%- endif -%}:&nbsp;
+  {%- for category in page.categories -%}
+    {%- if site.minima.archive_page.categories -%}
+      <a href="{{ site.minima.archive_page.categories }}/{{ category | slugify }}/">{{ category }}</a>
```
Comment (ashmaroli) [inline _includes/post-categories.html]: The intention here isn't clear.
What exactly is `site.minima.archive_page.categories` supposed to be? Is it just an arbitrary prefix in the URL?

Additionally, this doesn't respect the user's `baseurl` setting.

File: _includes/post-categories.html [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+    {%- else -%}
+      {{ category }}
+    {%- endif -%}
+    {%- if forloop.last == false -%},&nbsp;{%- endif -%}
```
Comment (ashmaroli) [inline _includes/post-categories.html]: use Liquid `unless` instead
```suggestion
    {% unless forloop.last %},&nbsp;{% endunless %}
```

File: _includes/post-categories.html [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
+  {%- endfor -%}
+</span>
```
Comment (ashmaroli) [inline _includes/post-categories.html]: We prefer having a trailing blank newline for ever file. Please configure your text-editor (if possible) or ensure an empty line at the end of file manually.
```suggestion
</span>

```

File: _includes/post-categories.html [hunk 4]
```diff
@@ -0,0 +1,10 @@
+<span class="post-categories">{%- if page.categories.size == 1 -%}Category{%- else -%}Categories{%- endif -%}:&nbsp;
```
[the remark this answers is not in the document]
Comment (hszhakka) [reply _includes/post-categories.html]: OK, I should use `{%` and `%}`.

File: _includes/post-categories.html (hunk 1 above)
Comment (hszhakka) [reply _includes/post-categories.html]: `site.minima.archive_page.categories` is a path to the categories page. 

Yes, it is arbitrary, but I should add the `baseurl`.

File: _includes/post-categories.html (hunk 2 above)
Comment (hszhakka) [reply _includes/post-categories.html]: OK

File: _includes/post-categories.html (hunk 3 above)
Comment (hszhakka) [reply _includes/post-categories.html]: OK, I have modified.

**Comment by ashmaroli** [comment]
Couple more edits to improve this:
- Prefer using the [`relative_url` filter](https://jekyllrb.com/docs/liquid/filters/) instead of prefixing `baseurl` manually. (You may have to use
`{% capture %}` in some places here..)
- We have deploy previews for every commit pushed. For example, the current state can be viewed at https://deploy-preview-423--jekyll-minima.netlify.com/. Use that to check if everything renders as expected.

File: .gitignore [hunk 5]
```diff
@@ -3,3 +3,4 @@
 Gemfile.lock
 _site
 *.gem
+.jekyll-cache
```
[the remark this answers is not in the document]
Comment (DirtyF) [reply .gitignore]: No, I don't have proper rights to do so, master branch is protected.

[the remark this answers is not in the document]
Comment (ashmaroli) [reply .gitignore]: Okay then. @hszhakka please add trailing newline to this file as well.
Thanks.

**Comment by hszhakka** [comment]
> Couple more edits to improve this:
> 
> * Prefer using the [`relative_url` filter](https://jekyllrb.com/docs/liquid/filters/) instead of prefixing `baseurl` manually. (You may have to use
>   `{% capture %}` in some places here..)
> * We have deploy previews for every commit pushed. For example, the current state can be viewed at https://deploy-preview-423--jekyll-minima.netlify.com/. Use that to check if everything renders as expected.

- I have used `relative_url` to replace `baseurl`.
- Yes, everything in that deployment is as expected.

**Comment by ashmaroli** [comment]
> * Yes, everything in that deployment is as expected.

Then why is it that I get a 404 from accessing the links at https://deploy-preview-423--jekyll-minima.netlify.com/categories/ ?

**Comment by hszhakka** [comment]
> > * Yes, everything in that deployment is as expected.
> 
> Then why is it that I get a 404 from accessing the links at https://deploy-preview-423--jekyll-minima.netlify.com/categories/ ?

Each link at that page links to a respective category page. Since every blog has different categories, the creation of these pages is the job of the user. I have provided a layout `_layouts/category.html`. For example, the category page for `misc` should look like this:

```
---
layout: category
category: misc
title: "Category: misc"
---
```

**Comment by ashmaroli** [comment]
So, if there are 10 categories in a *site*, the user has to create 10 different pages with just front matter..? That is not good UX...

Is that a Jekyll limitation?
In any case, do add those pages so that previews do not error with 404.

And finally, please document all this appropriately.

**Comment by hszhakka** [comment]
> So, if there are 10 categories in a _site_, the user has to create 10 different pages with just front matter..? That is not good UX...
> 
> Is that a Jekyll limitation?
> In any case, do add those pages so that previews do not error with 404.
> 
> And finally, please document all this appropriately.

Sure, writing 10 even 100 category pages manually is sucked, but the user can write a script to automatically generate all those files. I will upload the script later. 

I will write a complete document for this PR.

**Comment by ashmaroli** [comment]
> I will upload the script later.

Please don't do that. Scripts have no place in a theme-gem. I'll look into if an alternative is possible and update this branch.

**Comment by DirtyF** [comment]
I don't think we'll be able to merge this as Jekyll default pagination plugin don't support categories and tags, you'd have to use jekyll-pagination-v2 plugin that is not supported by GItHub Pages. It's nice to see people willing to improve Minima, but it's by design that this theme does not offer much out of the box, we're limited to default plugins.

**Comment by ashmaroli** [comment]
You may want to look into [`jekyll-archives`](https://github.com/jekyll/jekyll-archives) project for similar usage.

**Comment by hszhakka** [comment]
After some experiments, I feel this feature is not very elegant and also make Minima not friendly for beginners. It's better to keep this feature for personal use. So welcome to close this PR.

**Pull request closed without merging**